### DISCUSSION TITLE
Migration to Java 11 and Jakarta EE

### DISCUSSION THREAD

**Discussion by zubri** [prowide/prowide-iso20022#79] [Ideas]
I start this topic to talk about the migration that we are carrying out, to get some community users feedback as well on this regard.

From @essobedo PR initiative:

# Motivation

Many frameworks like Spring Boot, Quarkus, Camel... have made the Jakarta EE 10 migration which makes the lib incompatible with them. The goal of these changes is to migrate the lib to make it compatible again with the frameworks.

**Comment by zubri** [comment]
After announcing the migration plans via mailing to a broad list of users we started to get some feedback. 

Several users have constraints for the migration. We know already some of them cannot do it until the end of 2024. Others do not have complaints about Jakarta but will need Java 8 compatible version for a longer time.

So the idea is to continue with the release, but we should keep the migration to the minimal. In this sense, I think Jakarta could be migrated to EE 9 (XML bindings 3.0) instead of Jakarta EE 10. Spring Boot 3 is based on Jakarta EE 9.

Regarding a possible split of the migration between Java 11 and Jakarta that does not seem to be necessary. Anyway we will need to maintain the Java 8 Jaxb version for at least a couple of years.

@essobedo @ptorres-prowide

**Reply by abhimanyuchugh** [reply]
I created the ticket for the JAXB Upgrade Plan https://github.com/prowide/prowide-iso20022/issues/56.

Thank you for holding this discussion.

We are currently using Java 17, so we would welcome the upgrade to newer Jakarta EE, which would also unblock us from upgrading another internal project to Jakarta XML BInding 4.0.

From your comment, I assume you plan to maintain/release both Java 8 and Java 11 variants? Do you have any timeline for when a Java 11 version might be published?

Thanks!

**Comment by essobedo** [comment]
Are you sure that Spring Boot 3 is based on Jakarta EE 9? I agree that it is not clear, If we read [this very old blog post ](https://spring.io/blog/2021/09/02/a-java-17-and-jakarta-ee-9-baseline-for-spring-framework-6), we can assume so, but if we read [the release note](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Release-Notes#third-party-library-upgrades) and check [the code](https://github.com/spring-projects/spring-boot/blob/v3.0.0/spring-boot-project/spring-boot-dependencies/build.gradle#L554-L560), it seems that thery have migrated to XML binding 4.0 which is part of Jakarta EE 10, don't you agree?

**Reply by zubri** [reply]
Oh, you are right. They've changed afterwards.

**Reply by essobedo** [reply]
So what is your final decision regarding the target versions of Java/Jakarta? Still 8/9 or 11/10?

**Reply by zubri** [reply]
The library should be compatible with Spring Boot 3 so that implies at minimum 11/10 as you originally proposed.

**Comment by zubri** [comment]
RC1 has been released

https://central.sonatype.com/artifact/com.prowidesoftware/pw-iso20022/SRU2022-10.0.0-RC1

Including:
* Migration to Java 11
* Migration to Jakarta EE 10
* XmlGregorianCalendar was replaced model-wide by the new java.time classes

The corresponding Prowide Core dependency has also been release as RC1 with analogous changes:

https://central.sonatype.com/artifact/com.prowidesoftware/pw-swift-core/SRU2022-10.0.0-RC1

**Comment by ptorres-prowide** [comment]
Delivered. The library has been on Java 11 + Jakarta EE 10 (JAXB 4.0.x) since the 10.x line — XMLGregorianCalendar was replaced with java.time and there is no remaining javax.xml.bind usage. We also added optional test runs on Java 17 and 21. Closing as completed.
